### DISCUSSION TITLE
"Resource not accessible by integration" error on repository delete using user account token

### DISCUSSION THREAD

**Discussion by hyzyla** [community/community#72787] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

Hello! I'm trying to build application for managing github repositories. I've created Github application with "Administration" repository permissions  (page **Settings / Developer settings / GitHub Apps / [Projects cleaner](https://github.com/apps/projects-cleaner) / Permissions**):

<details><summary>Permission page screenshot</summary>
<img src="https://github.com/community/community/assets/7952949/7c1e5124-523d-4307-a4d9-256216f893d7">screenshot</img>
</details> 

I'm trying to authenticating with a GitHub App on behalf of a user and after authentification using NextAuth and client credentials for that app, I received access token, that starts with **ghu_** prefix. But when I try to make request to API to delete repository I get error "Resource not accessible by integration":

```sh
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer ghu_<TOKEN_HERE>" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  -X DELETE https://api.github.com/repos/hyzyla/gpt4all 
{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/repos/repos#delete-a-repository"
}
```

At the same time I can read repository via API using the same token:

```sh
curl -L \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer ghu_<TOKEN_HERE>" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  -X GET https://api.github.com/repos/hyzyla/gpt4all
{
  "id": 711137220,
  "node_id": "R_kgDOKmMXxA",
  "name": "gpt4all",
 ...
```

<details><summary>Shell screenshot</summary>
<img src="https://github.com/community/community/assets/7952949/25a81be8-1d45-4f30-bee7-d30e00c978eb">screenshot</img>
</details> 

Could you help me to find why I can't delete repository using API and user personal token?

**Comment by ghostinhershell** [comment]
Hey there @hyzyla, 

Thanks so much for reaching out to our community. I'm inclined to believe that the problem here is that the GitHub App does not have the necessary scopes. To delete a repository, you need to make sure that [the GitHub app has `write` access to Administration](https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#organization-permissions-for-administration). Could you please add that permission to the App and let me know if doing that solves the issue?

**Reply by hyzyla** [reply]
If I use Github Apps and my users login into my service using "Sign in with Github",  should I promt user additionally install app? or is it possilble to request auth & install app in one step?
